sdl_android/src/main/java/com/smartdevicelink/api/SdlManager.java

```diff
@@ -298,34 +302,50 @@ public Builder setContext(Context context){
 			return this;
 		}
 
-		public SdlManager build() {
-			if (sdlManager.appName == null) {
-				throw new IllegalArgumentException("You must specify an app name by calling setAppName");
-			}
+	public SdlManager build() {
+		if (sdlManager.appName == null) {
+			throw new IllegalArgumentException("You must specify an app name by calling setAppName");
+		}
 
-			if (sdlManager.appId == null) {
-				throw new IllegalArgumentException("You must specify an app ID by calling setAppId");
-			}
+		if (sdlManager.appId == null) {
+			throw new IllegalArgumentException("You must specify an app ID by calling setAppId");
+		}
 
-			if (sdlManager.hmiTypes == null) {
-				Vector<AppHMIType> hmiTypesDefault = new Vector<>();
-				hmiTypesDefault.add(AppHMIType.DEFAULT);
-				sdlManager.hmiTypes = hmiTypesDefault;
-				sdlManager.isMediaApp = false;
-			}
+		if (sdlManager.context == null) {
+			throw new IllegalArgumentException("You need to set context by calling setContext()");
+		}
 
-			if (sdlManager.hmiLanguage == null){
-				sdlManager.hmiLanguage = Language.EN_US;
-			}
+		if (sdlManager.hmiTypes == null) {
+			Vector<AppHMIType> hmiTypesDefault = new Vector<>();
+			hmiTypesDefault.add(AppHMIType.DEFAULT);
+			sdlManager.hmiTypes = hmiTypesDefault;
+			sdlManager.isMediaApp = false;
+		}
 
-			sdlManager.state = BaseSubManager.SETTING_UP;
+		if (sdlManager.lockScreenConfig == null){
+			// if lock screen params are not set, use default
+			LockScreenConfig lsc = new LockScreenConfig();
+			lsc.setEnabled(true);
+			lsc.setShowOEMLogo(false);
```

Comment:
Shouldn't these simply be set as default values instead of having to call set methods to create a default instantiation of the class?